citystatmodel.cpp

```diff
@@ -0,0 +1,207 @@
+//{
+//    if (value != headerData(section, orientation, role)) {
+//        // FIXME: Implement me!
+//        emit headerDataChanged(orientation, section, section);
+//        return true;
+//    }
+//    return false;
+//}
+
+
+int CityStatModel::rowCount(const QModelIndex &parent) const
+{
+    Q_UNUSED(parent);
+    return m_cityItemsList.size();
+}
+
+int CityStatModel::columnCount(const QModelIndex &parent) const
+{
+    Q_UNUSED(parent);
+    return 6;
+}
+
+QVariant CityStatModel::data(const QModelIndex &index, int role) const
+{
+    if (!index.isValid())
+        return QVariant();
+
+    if (index.row() >= m_cityItemsList.size() || index.row() < 0)
+         return QVariant();
+
+    QVariant returnValue;
+    if (role == Qt::DisplayRole || role == Qt::EditRole)
+    {
+        CityItem item = m_cityItemsList.at(index.row());
+        if (index.column() == 0)
+            returnValue = item.continentName;
+        else if (index.column() == 1)
+            returnValue = item.countryName;
+        else if (index.column() == 2)
+            returnValue = item.cityName;
+        else if (index.column() == 3)
+            returnValue = item.cityPopulation;
+        else if (index.column() == 4)
+            returnValue = item.recordHighTempereture;
+        else if (index.column() == 5)
+            returnValue = item.recordLowTempereture;
+     }
+     return returnValue;
+}
+
+bool CityStatModel::setData(const QModelIndex &index, const QVariant &value, int role)
+{
+    if (data(index, role) != value)
+    {
+        switch (index.column())
+        {
+        case 0:
+        {
+            m_cityItemsList[index.row()].continentName=value.toString();
+            break;
+        }
+        case 1:
+        {
+            m_cityItemsList[index.row()].countryName=value.toString();
+            break;
+        }
+        case 2:
+        {
+            m_cityItemsList[index.row()].cityName=value.toString();
+            break;
+        }
+        case 3:
+        {
+            m_cityItemsList[index.row()].cityPopulation=value.toUInt();
+            fillPopulationInfo();
+            break;
+        }
+        case 4:
+        {
+            m_cityItemsList[index.row()].recordHighTempereture=value.toInt();
+            break;
+        }
+        case 5:
+        {
+            m_cityItemsList[index.row()].recordLowTempereture=value.toInt();
+            break;
+        }
+        }
+        emit dataChanged(index, index, QVector<int>() << role);
+        return true;
+    }
+    return false;
+}
+
+Qt::ItemFlags CityStatModel::flags(const QModelIndex &index) const
+{
+    return QAbstractTableModel::flags(index) | Qt::ItemIsEditable;
+}
+
+bool CityStatModel::insertRows(int row, int count, const QModelIndex &parent)
+{
+    beginInsertRows(parent, row, row + count - 1);
+    for (int i = 0; i<count;i++)
+    {
+        CityItem item;
+        item.continentName="";
```

Comment:
This should be a part of CityItem default ctor

Reply:
ok, cleared